habitatstreams/src/21_update_sample_frame_3260.Rmd

```diff
@@ -0,0 +1,405 @@
+
+- Finally we assign a grts-ranking to each vha point based on to the grts-master
+
+```{r}
+map_3260_old_strata <- map_3260_old %>%
+  mutate(sac_old = ifelse(SBZ == "sa", "downstream",
+                          ifelse(SBZ == "intersect SBZ", "intersect",
+                                 ifelse(SBZ == "buiten", "outside", NA)))) %>%
+  dplyr::select(sac_old, naam) %>%
+  st_buffer(dist = 1) 
+  # st_join(sac, largest = TRUE) %>%
+  # rename(sac_code_segment = sac_code)
+
+
+vha_segments_combined <- segments_3260_old_sf %>%
+  st_join(sac) %>%
+  st_join(map_3260_old_strata, largest = TRUE) %>%
+  mutate(id = str_c("old_", segment_id),
+         x = st_coordinates(.)[,1],
+         y = st_coordinates(.)[,2]) %>%
+  st_drop_geometry() %>%
+  dplyr::select(id, name = river_name, wl_code, wl_code_old, sac_code, sac_old, grts_ranking, x, y) %>%
+  bind_rows(vha_difference_buffer)
+  
+#vha_segments_combined %>%
+  # st_as_sf(coords = c("x", "y"), crs = 31370) %>%
+  # st_write(dsn = "../output/check2.gpkg", "vha_segments_combined", driver ="GPKG")
+```
+
+## Check if points are located within Flanders and overlap with grts-master  
+
+
+```{r}
+
+flanders <- read_admin_areas()
+
+streams_segments_no_ranking <- vha_segments_combined %>%
+  rename(river_name = name) %>%
+  filter(is.na(grts_ranking)) %>%
+  st_as_sf(coords = c("x", "y"), crs = 31370) %>%
+  st_join(flanders) %>%
+  mutate(in_flanders = !is.na(name),
+         color_in_flanders = ifelse(in_flanders, "blue", "red"))
+
+overview_missing_ranking <-  vha_segments_combined %>%
+    rename(river_name = name) %>%
+  st_as_sf(coords = c("x", "y"), crs = 31370) %>%
+  st_join(flanders) %>%
+  st_drop_geometry() %>%
+  mutate(in_flanders = !is.na(name),
+         in_grts_master = !is.na(grts_ranking)) %>%
+  group_by(in_flanders, in_grts_master) %>%
+  summarise(n = n()) %>%
+  ungroup()
+
+grts_border <- grts_master %>%
+  mask(st_buffer(streams_segments_no_ranking, 32)) %>%
+  rasterToPolygons() %>%
+  st_as_sf()
+
+n_flanders_missing <- (overview_missing_ranking %>%
+  filter(in_flanders, !in_grts_master))$n
+
+n_flanders_tot <- (overview_missing_ranking %>%
+  filter(in_flanders))$n %>%
+  sum()
+
+percent_flanders_missing <- round(n_flanders_missing/n_flanders_tot * 100, 2)
+
+```
+
+The table below gives an overview of the location of the vha points. `r n_flanders_missing` vha points are located within Flanders but do not overlap with the grts master. 
+This is because the grts master only contains cells for which the centroid is located within Flanders. 
+Therefore some parts of Flanders at the borders are not covered by the grts master. 
+However, only `r percent_flanders_missing` % of vha points within Flanders do not overlap with the grts master.     
+
+```{r}
+overview_missing_ranking %>%
+  kable() %>%
+  kable_styling()
+```
```

Comment:
Given below table, it is remarkable that points outside Flanders exist. I guess this must be due to the VHA exceeding Flemish borders?

in_flanders | in_grts_master | n
-- | -- | --
FALSE | FALSE | 627
FALSE | TRUE | 790
TRUE | FALSE | 301
TRUE | TRUE | 236479